src/core/capabilities/capabilityAggregator.ts

```
@@ -241,13 +246,12 @@ export class CapabilityAggregator extends EventEmitter {
-  private async safeListTools(
-    serverName: string,
-    client: Pick<Client, 'listTools'>,
-    transport: EnhancedTransport,
-  ): Promise<ListToolsResult> {
+  private async safeListTools(serverName: string, connection: OutboundConnection): Promise<ListToolsResult> {
     try {
-      return await client.listTools(undefined, { timeout: getRequestTimeout(transport) });
+      return await executeWithPostAuthOAuthRecovery(serverName, connection, () => connection.client.listTools());
+      return await executeWithPostAuthOAuthRecovery(serverName, connection, () =>
+        connection.client.listTools(undefined, { timeout: getRequestTimeout(connection.transport as EnhancedTransport) }),
+      );
```

ESLint

[error] Unreachable code.

(no-unreachable)